src/compiler/checker.ts

```diff
@@ -10606,6 +10609,29 @@ namespace ts {
             return false;
         }
 
+        function elaborateDidYouMeanToCallOrConstruct(node: Expression, source: Type, target: Type): boolean {
+            const callSignatures = getSignaturesOfType(source, SignatureKind.Call);
+            const constructSignatures = getSignaturesOfType(source, SignatureKind.Construct);
+            for (const signatures of [constructSignatures, callSignatures]) {
+                if (length(signatures)) {
+                    if (some(signatures, s => {
+                        const returnType = getReturnTypeOfSignature(s);
+                        return !(returnType.flags & (TypeFlags.Any | TypeFlags.Never)) && isTypeAssignableTo(returnType, target);
+                    })) {
+                        const resultObj: { error?: Diagnostic } = {};
+                        checkTypeAssignableTo(source, target, node, /*errorMessage*/ undefined, /*containingChain*/ undefined, resultObj);

```

Comment:
TIL `errorOutputContainer`

Reply:
Yeah, I added that awhile ago. Not the happiest about it, but a side channel's a side channel, since most callers just need the boolean return.